libs/dotnet-sdk/FeatureBoardHttpClient.cs

```diff
@@ -46,19 +52,33 @@ public FeatureBoardHttpClient(HttpClient httpClient, LastETagProvider lastModifi
         return null;
     }
 
-    var features = await response.Content.ReadFromJsonAsync<List<FeatureConfiguration>>(cancellationToken: cancellationToken)
+      features = await response.Content.ReadFromJsonAsync<List<FeatureConfiguration>>(cancellationToken: cancellationToken)
                     ?? throw new ApplicationException("Unable to retrieve decode response content");
 
-    _processResult(features);
     updateEtagRef(response.Headers.ETag);
+    }
+    catch (HttpRequestException e)
+    {
+      _logger.LogError(e, "Failed to get latest flags");
+      return null;
+    }
+
+    try
+    {
+      await OnFeatureConfigurationUpdated(features, cancellationToken);
+      return true;
+    }
+    catch (ArgumentException e) // eg. thrown due to duplicate feature key
+    {
+      _logger.LogError(e, "Failed to update flags");
```

Comment:
Should the eTag value be rolled back to its original value if exception occurs at this point (ie. during handling of the `FeatureConfigurationUpdated` event)?

Reply:
I am not hugely experienced in handling eTags, but isn't just using a new value fine since you are trying to compare tag values in the end?

Reply:
If the eTag retains its updated value and isn't rolled back then the code inside try block won't run again until FeatureBoard API responds back with a non 304 status (due to someone touching the feature config). 

I think, at least for this exception scenario [caused by duplicate feature name appearing in FB API response), rollback is appropriate, so it keeps retrying the state update periodically.

Reply:
Tricker question is what to do if `IFeatureBoardExternalState` throws